src/test/java/org/checkstyle/autofix/recipe/AbstractRecipeTestSupport.java

```diff
@@ -63,38 +64,58 @@ protected void verify(String testCaseName) throws Exception {
         final Configuration config = getCheckConfigurations(inputPath);
         verifyOutputFile(outputPath, config);
 
-        final List<CheckstyleViolation> violations = runCheckstyle(inputPath, config);
+        final Path violations = runCheckstyle(inputPath, config);
 
-        verifyWithInlineConfigParser(getPath(inputPath), convertToExpectedMessages(violations));
+        verifyWithInlineConfigParser(getPath(inputPath),
+                convertToExpectedMessages(CheckstyleReportParser.parse(violations)));
 
         final String beforeCode = readFile(getPath(inputPath));
         final String expectedAfterCode = readFile(getPath(outputPath));
         final CheckConfiguration checkConfig = ConfigurationLoader.mapConfiguration(config);
-        final Recipe mainRecipe = createRecipe(violations, checkConfig);
-        testRecipe(beforeCode, expectedAfterCode,
-                getPath(inputPath), new InputClassRenamer(),
-                mainRecipe, new RemoveViolationComments());
+        final Path configPath = Files.createTempFile("checkstyle-config", ".xml");
+
+        try {
+            writeConfigurationToXml(checkConfig, configPath.toString());
+            final Recipe mainRecipe = new CheckstyleAutoFix(violations.toString(),
+                    configPath.toString());
```

Comment:
you should probably use a different method, `toString()` on Path may not be reliable. Please check alternatives